### PR TITLE
Add an introspection endpoint for container metadata

### PR DESCRIPTION
**Pull request by Molter73**
## Description

Add an introspection endpoint to collector which can be used to retrieve container metadata by providing it with a container ID. At the time, the only data that is returned by the endpoint is the same container ID that was provided and the k8s namespace the container belongs to. This endpoint will enable testing of the #1606 PR without need for changes to our communication with sensor.

## Checklist
- [x] Investigated and inspected CI test results
- [x] Updated documentation accordingly

**Automated testing**
  - [ ] Added unit tests
  - [x] Added integration tests
  - [ ] Added regression tests

If any of these don't apply, please comment below.

## Testing Performed

Manually tested the new endpoint.

### CODE REVIEW

File: collector/lib/IntrospectionEndpoint.h [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+#include <unordered_map>
+
+namespace collector {
+
+using QueryParams = std::unordered_map<std::string, std::string>;
+
+class IntrospectionEndpoint : public CivetHandler {
+ protected:
+  static QueryParams ParseParameters(const char* queryString);
+  static std::optional<std::string> GetParameter(const QueryParams& params, const std::string& paramName);
```
Comment (ovalenti) [inline collector/lib/IntrospectionEndpoint.h:15]: These methods are not used in the PR. However, I think they will be soon, as we add more endpoints.

**Review by ovalenti — APPROVED** [review]
Looks good, and thanks for extracting the IntrospectionEndpoint common functions !

My only concern is with the underlying concurrent access via GetNamespace(). I left a comment in the other PR.

File: docs/troubleshooting.md [hunk 2]
```diff
@@ -450,3 +450,86 @@ $ curl collector:8080/profile/heap
...
+80bfaff4d03b
+```
+
+You can then port-forward the collector daemonset to be able to query the
+container metadata endpoint from your localhost. Alternatively, you can exec
+into a pod with access to an HTTP tool in order to query the endpoint from
+within the cluster itself.
+
+```sh
+$ krox port-forward ds/collector 8080:8080 &
```
Comment (ovalenti) [inline docs/troubleshooting.md:528]: It looks like the initial part of the command line is missing.

File: docs/troubleshooting.md [hunk 3]
```diff
@@ -450,3 +450,86 @@ $ curl collector:8080/profile/heap
...
+```sh
+$ kubectl -n stackrox get pods -l "app=collector" -o json \
+    | jq -r '.items[].status.containerStatuses[].containerID' \
+    | sed -e 's#containerd://##' \
+    | cut -c -12
+01e8c0454972
+80bfaff4d03b
+```
+
+You can then port-forward the collector daemonset to be able to query the
```
Comment (ovalenti) [inline docs/troubleshooting.md:522]: If we are going to have more endpoints, this port-forward hint could be placed in the common section a bit above.

File: docs/troubleshooting.md (hunk 2 above)
Comment (Molter73) [reply docs/troubleshooting.md:528]: Dang, `krox` is an alias I have for `kubectl -n stackrox`, I'll get it fixed.

File: docs/troubleshooting.md (hunk 3 above)
Comment (Molter73) [reply docs/troubleshooting.md:522]: Sounds good, I'll move it to the common section.

**Review by ovalenti — APPROVED** [review]
Still looks good to me :+1:

**Pull request merged**